src/beacons/use-on-beacons-session-count.tsx

```diff
@@ -0,0 +1,78 @@
+import {useOnResume} from '@atb/utils/use-on-resume';
+import {useEffect, useState} from 'react';
+import {useIsBeaconsEnabled} from './use-is-beacons-enabled';
+import {storage} from '@atb/storage';
+
+export const useOnBeaconsSessionCount = (
+  count: number,
+  runOnCountFunc: () => void,
+  keyAppendix: string, // appendix to ensure unique storage keys
+) => {
+  const beaconsSessionCount = useBeaconsSessionCounter(count + 1, keyAppendix);
+
+  useEffect(() => {
+    if (beaconsSessionCount !== undefined && beaconsSessionCount === count) {
+      runOnCountFunc();
+    }
+  }, [beaconsSessionCount]);
+};
+
+enum storeKey {
+  beaconsSessionCountKey = '@ATB_session_count_',
```

Comment:
Is this ment to end with `_ `?

Reply:
Yes. As `keyAppendix` is set to `"share_travel_habits"`, the full key will be `"@ATB_session_count_ share_travel_habits"` in this case.